controllers/replace_failed_pods_test.go

```diff
@@ -49,63 +52,90 @@ var _ = Describe("replace_failed_pods", func() {
 		Expect(generation).To(Equal(int64(1)))
 	})
 
-	Describe("chooseNewRemovals", func() {
-		var result bool
-		JustBeforeEach(func() {
-			result = chooseNewRemovals(cluster)
+	JustBeforeEach(func() {
+		result = chooseNewRemovals(clusterReconciler, cluster)
+	})
+
+	Context("with no missing processes", func() {
+		It("should return false", func() {
+			Expect(result).To(BeFalse())
+		})
+
+		It("should not mark anything for removal", func() {
+			Expect(getRemovedProcessGroupIDs(cluster)).To(Equal([]string{}))
+		})
+	})
+
+	Context("with a process that has been missing for a long time", func() {
+		BeforeEach(func() {
+			processGroup := fdbtypes.FindProcessGroupByID(cluster.Status.ProcessGroups, "storage-2")
+			processGroup.ProcessGroupConditions = append(processGroup.ProcessGroupConditions, &fdbtypes.ProcessGroupCondition{
+				ProcessGroupConditionType: fdbtypes.MissingProcesses,
+				Timestamp:                 time.Now().Add(-1 * time.Hour).Unix(),
+			})
 		})
 
-		Context("with no missing processes", func() {
+		Context("with no other removals", func() {
 			It("should return false", func() {
-				Expect(result).To(BeFalse())
+				Expect(result).To(BeTrue())
```

Comment:
This condition seems like it's no longer matching the `It` statement above it. Does that need a change as well?

Reply:
Good catch I'll check that test case. I assume fat fingers when I moved the test cases outside the second describe block.